source/server/http_filter_config_base.cc

```diff
@@ -13,11 +13,11 @@ FilterConfigurationBase::FilterConfigurationBase(
 
 void FilterConfigurationBase::computeEffectiveConfiguration(
     const Envoy::Http::RequestHeaderMap& headers) {
-  const auto* request_config_header = headers.get(TestServer::HeaderNames::get().TestServerConfig);
-  if (request_config_header) {
+  const auto& request_config_header = headers.get(TestServer::HeaderNames::get().TestServerConfig);
+  if (request_config_header.size() == 1) {
```

Comment:
should we have an else condition here now?  again, taking into consideration receiving more than 1

Reply:
Yes, you're right. Done in dc0b618951bde50cc9eed7143fe39d0f7d433521 + explicit tests for the case.